src/pages/MainPage/components/AddTodo/useAddTodoToDb.ts

```diff
@@ -0,0 +1,42 @@
+import { useState } from "react";
+import { collection, addDoc } from "firebase/firestore";
+import db from "../../../../backend/firebase";
+import { checkTodo } from "./AddTodo.utils";
+
+interface IStatus {
+  added: boolean;
+  errors: string[];
+}
+
+const useAddTodoToDb = (todoDetails: ITodoDetails) => {
+  const [loading, setLoading] = useState(false);
+  const [status, setStatus] = useState<IStatus>({ added: false, errors: [] });
+
+  const submitTodo = async () => {
+    setStatus({ added: false, errors: [] });
+    if (checkTodo(todoDetails).length > 0) {
```

Comment:
w sumie 2 razy wywołujesz tę samą funkcję po te same dane, możesz po prostu nad ifem zrobić:
 const errors = checkTodo(todoDetails)
 
 i wtedy sprawdzić w ifie czy to jest większe od zera i wykorzystać to też w setStatus